src/gtk/app.cpp

```diff
@@ -178,6 +181,35 @@ bool wxApp::DoIdle()
     return keepSource;
 }
 
+// Custom Glib log writer
+static GLogWriterOutput wx_log_writer(GLogLevelFlags   log_level,
+                                      const GLogField *fields,
+                                      gsize            n_fields,
+                                      gpointer         user_data)
+{
+    const wxUIntPtr log_mask = reinterpret_cast<wxUIntPtr>(user_data);
+
+    GLogWriterOutput result;
+    if (log_level & log_mask)
+    {
+        result = G_LOG_WRITER_HANDLED;
+    }
+    else
+    {
+        result = g_log_writer_default(log_level, fields, n_fields, NULL);
+    }
+    return result;
+}
+
+/* static */
+void wxApp::GTKSuppressDiagnostics(long flags)
+{
+    g_log_set_writer_func(
```

Comment:
`g_log_set_writer_func()` requires GLib 2.50

Reply:
Added checks, thanks for noticing this!